src/dispatch/case/flows.py

```diff
@@ -660,20 +623,23 @@ def case_to_incident_endpoint_escalate_flow(
     case.status = CaseStatus.escalated
 
     # we run the incident create flow
-    incident = incident_flows.incident_create_flow(
+    incident_flows.incident_create_flow(
         incident_id=incident_id, organization_slug=organization_slug, db_session=db_session
     )
-    case.incidents.append(incident)
 
-    db_session.add(case)
-    db_session.commit()
+    # fetch the incident from the database after the background task has completed
+    incident = incident_service.get(db_session=db_session, incident_id=incident_id)
+    if incident:
```

Comment:
```suggestion
    if incident := incident_service.get(db_session=db_session, incident_id=incident_id):
```

Use walrus operator as this var is used nowhere else